common/network-common/src/main/java/org/apache/spark/network/client/TransportClientFactory.java

```diff
@@ -210,6 +210,14 @@ private TransportClient createClient(InetSocketAddress address)
       .option(ChannelOption.CONNECT_TIMEOUT_MILLIS, conf.connectionTimeoutMs())
       .option(ChannelOption.ALLOCATOR, pooledAllocator);
 
+    if (conf.receiveBuf() > 0) {
+      bootstrap.option(ChannelOption.SO_RCVBUF, conf.receiveBuf());
```

Comment:
For server side, netty leverages reactor pattern and `ServerBootstrap` is used here, method `option` works when socket binding and connecting occurs, while `childOption` works for the later process when connection is established and there will be one channel per client.
For client side, spark rpc uses `Bootstrap`, there is no method to set `childOption` because client works not like server side, it is pretty simple for client, only one thread-pool will be used by netty to do network I/O.